services/utilServices/AuthService.js

```diff
@@ -0,0 +1,118 @@
+const axios = require("axios")
+const queryString = require("query-string")
+const uuid = require("uuid/v4")
+
+// Import error types
+const { AuthError } = require("@errors/AuthError")
+const { ForbiddenError } = require("@errors/ForbiddenError")
+
+const validateStatus = require("@utils/axios-utils")
+const jwtUtils = require("@utils/jwt-utils")
+
+const logger = require("@root/logger/logger")
+
+const { CLIENT_ID, CLIENT_SECRET, REDIRECT_URI } = process.env
+
+class AuthService {
+  constructor({ usersService }) {
+    this.usersService = usersService
+  }
+
+  async getAuthRedirectDetails() {
+    const state = uuid()
+    const githubAuthUrl = `https://github.com/login/oauth/authorize?client_id=${CLIENT_ID}&state=${state}&scope=repo`
+
+    const token = jwtUtils.signToken({ state })
+
+    return { redirectUrl: githubAuthUrl, cookieToken: token }
+  }
+
+  async getGithubAuthToken({ csrfState, code, state }) {
+    try {
+      const decoded = jwtUtils.verifyToken(csrfState)
+      if (decoded.state !== state) {
+        logger.error("The given github credentials are not authorized!")
+        throw new Error("State does not match")
+      }
+    } catch (err) {
+      // Transform jwt errors into generic ForbiddenErrors - the raw errors shouldn't be provided to the user
+      throw new ForbiddenError(
+        `csrfState: ${csrfState}, code: ${code}, state: ${state}`
```

Comment:
double checking if the user will see this; if they will, this should just be a `logger.error` statement. if they won't, this is fine. 

main concern here is that we want to avoid confusing the users, especially when the error message contains q technical information.

Reply:
Whoops, thanks for catching this! Fixed in https://github.com/isomerpages/isomercms-backend/pull/368/commits/62e348a631ebecf28b20447107f12d348ae33075